Spigot-Server-Patches/0267-PlayerAdvancementGrantEvent.patch

```diff
@@ -0,0 +1,26 @@
+From 766a43c8f6a21e23316c19fc1e9adfbeb1a9a878 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Fri, 19 Jan 2018 08:15:29 -0600
+Subject: [PATCH] PlayerAdvancementGrantEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/AdvancementDataPlayer.java b/src/main/java/net/minecraft/server/AdvancementDataPlayer.java
+index 6896b709..996f4796 100644
+--- a/src/main/java/net/minecraft/server/AdvancementDataPlayer.java
++++ b/src/main/java/net/minecraft/server/AdvancementDataPlayer.java
+@@ -196,6 +196,12 @@ public class AdvancementDataPlayer {
+         boolean flag1 = advancementprogress.isDone();
+ 
+         if (advancementprogress.a(s)) {
++            // Paper start
++            if (!new com.destroystokyo.paper.event.player.PlayerAdvancementGrantEvent(this.player.getBukkitEntity(), advancement.bukkit, s).callEvent()) {
```

Comment:
i think we should rename the event to PlayerAdvancementCriterionGrantedEvent
because the current suggested name implies its granting the Advancement itself, which Bukkit does already have an event for right below this once all criterion are satisfied.

Reply:
Done.